app/models/post.rb

```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes, foreign_key: 'post_id'
+
+  after_save :update_posts_counter
```

Comment:
Awesome 🚀 You used one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) to perform the posts counter 📝